File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/test/model/enumeration/QueueOperationTest.java
package com.digirati.elucidate.common.test.model.enumeration;

import org.junit.Test;

import com.digirati.elucidate.common.model.enumeration.QueueOperation;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertNotNull;

public class QueueOperationTest {

    @Test
    public void testQueueOperationCreate() {
        QueueOperation queueOperation = QueueOperation.CREATE;
        assertNotNull(queueOperation);
        assertThat(queueOperation, is(equalTo(QueueOperation.CREATE)));
    }

    @Test
    public void testQueueOperationUpdate() {
        QueueOperation queueOperation = QueueOperation.UPDATE;
        assertNotNull(queueOperation);
        assertThat(queueOperation, is(equalTo(QueueOperation.UPDATE)));
    }

    @Test
    public void testQueueOperationDelete() {
        QueueOperation queueOperation = QueueOperation.DELETE;
        assertNotNull(queueOperation);
        assertThat(queueOperation, is(equalTo(QueueOperation.DELETE)));
    }
}

File: elucidate-common-lib/src/test/kotlin/com/digirati/elucidate/common/test/model/enumeration/QueueOperationTest.kt
package com.digirati.elucidate.common.test.model.enumeration

import com.digirati.elucidate.common.model.enumeration.QueueOperation
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Assert
import org.junit.Test

class QueueOperationTest {
    @Test
    fun testQueueOperationCreate() {
        val queueOperation = QueueOperation.CREATE
        Assert.assertNotNull(queueOperation)
        MatcherAssert.assertThat(queueOperation, Matchers.`is`(Matchers.equalTo(QueueOperation.CREATE)))
    }

    @Test
    fun testQueueOperationUpdate() {
        val queueOperation = QueueOperation.UPDATE
        Assert.assertNotNull(queueOperation)
        MatcherAssert.assertThat(queueOperation, Matchers.`is`(Matchers.equalTo(QueueOperation.UPDATE)))
    }

    @Test
    fun testQueueOperationDelete() {
        val queueOperation = QueueOperation.DELETE
        Assert.assertNotNull(queueOperation)
        MatcherAssert.assertThat(queueOperation, Matchers.`is`(Matchers.equalTo(QueueOperation.DELETE)))
    }
}